ShopAppTests/App/Account/Controllers/AddressFormViewControllerSpec.swift

```diff
@@ -0,0 +1,277 @@
+            }
+            
+            context("if it has correct filled text field views") {
+                it("needs to enable submit button") {
+                    viewModelMock.makeSubmitButtonEnabled()
+                    expect(submitButton.isEnabled) == true
+                }
+            }
+        }
+        
+        describe("when submit button did press") {
+            var disposeBag: DisposeBag!
+            var delegateMock: AddressFormControllerlDelegateMock!
+            
+            beforeEach {
+                disposeBag = DisposeBag()
+                
+                viewController = UIStoryboard(name: StoryboardNames.account, bundle: nil).instantiateViewController(withIdentifier: ControllerIdentifiers.addressForm) as! AddressFormViewController
+                
+                let repositoryMock = PaymentsRepositoryMock()
+                let addressFormUseCaseMock = CountriesUseCase(repository: repositoryMock)
+                viewModelMock = AddressFormViewModelMock(countriesUseCase: addressFormUseCaseMock)
+                viewController.viewModel = viewModelMock
+                
+                delegateMock = AddressFormControllerlDelegateMock()
+                viewController.delegate = delegateMock
+                
+                countryPicker = self.findView(withAccessibilityLabel: "addressFormCountryPicker", in: viewController.view) as! BasePicker
+                nameTextFieldView = self.findView(withAccessibilityLabel: "addressFormName", in: viewController.view) as! InputTextFieldView
+                lastNameTextFieldView = self.findView(withAccessibilityLabel: "addressFormLastName", in: viewController.view) as! InputTextFieldView
+                addressTextFieldView = self.findView(withAccessibilityLabel: "addressFormAddress", in: viewController.view) as! InputTextFieldView
+                addressOptionalTextFieldView = self.findView(withAccessibilityLabel: "addressFormAddressOptional", in: viewController.view) as! InputTextFieldView
+                cityTextFieldView = self.findView(withAccessibilityLabel: "addressFormCity", in: viewController.view) as! InputTextFieldView
+                statePicker = self.findView(withAccessibilityLabel: "addressFormState", in: viewController.view) as! BasePicker
+                zipCodeTextFieldView = self.findView(withAccessibilityLabel: "addressFormZipCode", in: viewController.view) as! InputTextFieldView
+                phoneTextFieldView = self.findView(withAccessibilityLabel: "addressFormPhone", in: viewController.view) as! InputTextFieldView
+                submitButton = self.findView(withAccessibilityLabel: "addressFormSubmitButton", in: viewController.view) as! BlackButton
+            }
+            
+            it("should return correct address") {
+                viewModelMock.makeSubmitButtonEnabled()
+                
+                viewController.viewModel.filledAddress
+                    .subscribe(onNext: { address in
```

Comment:
Can we avoid subscription in test case and test real binding between view controller and view model?